etc/kayobe/ansible/prometheus.yml.j2

```diff
@@ -0,0 +1,287 @@
+{{ '{%' }} raw {{ '%}' }}
+{% raw %}
+global:
+  scrape_interval: {{ prometheus_scrape_interval }}
+  scrape_timeout: 10s
+  evaluation_interval: 15s
+{% if prometheus_external_labels %}
+  external_labels:
+{% for label, value in prometheus_external_labels.items() %}
+    {{ label }}: {{ value }}
+{% endfor %}
+{% endif %}
+
+{% if prometheus_alert_rules.files is defined and prometheus_alert_rules.files | length  > 0 %}
+rule_files:
+{% for rule in prometheus_alert_rules.files %}
+  - "/etc/prometheus/{{ rule.path | basename }}"
+{% endfor %}
+{% endif %}
+
+scrape_configs:
+  - job_name: prometheus
+    basic_auth:
+      username: admin
+      password: "{{ prometheus_password }}"
+    static_configs:
+{% for host in groups['prometheus'] %}
+      - targets:
+        - '{{ 'api' | kolla_address(host) | put_address_in_context('url') }}:{{ prometheus_port }}'
+{% if hostvars[host].prometheus_instance_label | default(false, true) %}
+        labels:
+          instance: "{{ hostvars[host].prometheus_instance_label }}"
+{% endif %}
+{% endfor %}
+
+{% if enable_prometheus_node_exporter | bool %}
+  - job_name: node
+    static_configs:
+{% for host in groups['prometheus-node-exporter'] %}
+      - targets:
+        - '{{ 'api' | kolla_address(host) | put_address_in_context('url') }}:{{ hostvars[host]['prometheus_node_exporter_port'] }}'
+{% if hostvars[host].prometheus_instance_label | default(false, true) %}
+        labels:
+          instance: "{{ hostvars[host].prometheus_instance_label }}"
+{% endif %}
+{% endfor %}
+{% endraw %}
+    metric_relabel_configs:
+     - replacement: $1
+       source_labels: [device]
+       target_label: 'original_device'
+{% for net_map in prometheus_network_maps_blob %}
```

Comment:
Kolla ansible does have access to at least some of this data, although the networks don't have quite the same names as those in kayobe config.

```
api_interface
storage_interface
tunnel_interface
```
etc.

Of course the value of these may vary between hosts.

Reply:
This was considered - the issue is that only some of the info is there as you point out.